[READY] tgui

neersighted opened the pull request:
Announcing... tgui!

NanoUI has now been rewritten, frontend and backend -- thus, it now has a new name.

This is hopefully the last (heh!) ever rewrite. At this point most of the work to be done is documentation and converting stringbuilder UIs.

It kind of changes a lot -- UIs are the same as before, differences are mostly cosmetic.
The immediate difference is more consistent styles and animated UIs (numbers and bars and buttons).
The animations are purely visual and I will probably add the option to disable them (the fancy pref) if people complain. They work great even on IE8 however.

This update should also fix compatibility with IE8; as I am now using a framework to keep things sane.

Mostly, this rewrite opens up many possibilities for future UIs, such as minimaps and graphs, as each UI can have code and styles specific to it; as well as breaking up large UIs into multiple files (Air Alarms are like this).

So with that, all hail the era of tgui!

:cl:
tweak: NanoUI is now tgui (aka NanoUI 4)
experiment: Rewrote NanoUI again... everything should be fixed; hopefully.
:cl:

neersighted committed:
Begin Ractive rewrite of NanoUI

* Switch JS builds to browserify
* Rip out doT; modularize Gulp build code
* Move scripts and templates to old folders; get a "Hello World" demo going

GunHog commented:
At least provide a log of what you did.

optimumtact commented:
read the code

KazeEspada commented on nano/gulp/tasks/js.coffee:
```diff
@@ -0,0 +1,41 @@
+
+
+browserify = require "browserify"
+gulp       = require "gulp"
```
GULP!

Shadowlight213 commented:
@optimumtact 
>normal coders being able to read nanoui code

We aren't that good.

neersighted committed:
More progress on the Ractive rewrite

* Get APCs working under the new system; clean up code
* Auto-update fancy settings; remove README (outdated now); clean Titlebar
* Convert to Stylus; seperate components and such

neersighted committed:
Attempt to reuse DOM elements in templates

Also, remove unused/extra styles to save space (ha ha ha)

neersighted committed:
Add error interface; clean animations; use webfont loader; package JSON better

neersighted committed:
Clean execution order; make interface style reactive

Add Stylus compiler to components; this allows stylus in components

neersighted committed:
Add watch task

Also, fix link transitions and tidy up comments and helper files;
shuffle CSS, refine component event handling

neersighted committed:
Revise math extensions

neersighted committed:
Add interfaces

* Add tanks interface
* Add canister interface
* Add chem heater interface; wrap long template lines
* Add chem dispenser interface
* Add section header buttons

neersighted committed:
Re-design styles to be more consistant and be a table

Also add transitions to chemistry things

neersighted committed:
Remove old scripts; done with these

neersighted committed:
Add cryo interface

neersighted commented:
Just gonna put this here...

![2015-12-31_17-58-51](https://cloud.githubusercontent.com/assets/1049222/12069188/ab1a07f4-afe8-11e5-8971-faa53ca052a9.gif)

neersighted commented:
Also for comparison with OG nanoui:
![2015-11-30_21-06-14](https://cloud.githubusercontent.com/assets/1049222/12069209/5e406184-afe9-11e5-903c-2c86b0c5e64c.png)

neersighted committed:
Fix buttons in headers; use computed data

neersighted committed:
Add more complete UIs

* Add airlock electronics UI
* Add space heater UI
* Add SMES interface

neersighted committed:
Use NanoUI with your keyboard (tab and enter)!

neersighted committed:
Finish even more UIs...

* Add atmos UIs and solar interface
* Add air alarm interface

Shadowlight213 commented:
Oh my god that new updating is amazing. :+1:

neersighted committed:
Fix IE8 support; Factor out some code into the NanoUI component

neersighted committed:
NanoUI is now 'tgui'

neersighted committed:
Rename the tgui pref and adjust all UI sizes for the rewrite

neersighted commented:
It happening!!!!
